Wrappers/Python/ccpi/viewer/CILViewer.py

```diff
@@ -712,52 +726,264 @@ def addHeadlight(self):
+
+    def setScalarOpacityPercentiles(self, min, max, update_pipeline=True):
+        '''
+        min, max: float, default: (80., 99.)
+            the percentiles on the image values that the 
+            opacity will be mapped to if setVolumeRenderOpacityMethod
+            has been set to 'scalar'.
+        '''
+        so_min, so_max = self.getVolumeMapWindow((min, max), 'scalar')
+        self.setScalarOpacityWindow(so_min, so_max, update_pipeline)
+
+    def setVolumeColorPercentiles(self, min, max, update_pipeline=True):
+        '''
+        min, max: int, default: (85., 95.)
+            the percentiles on the image values upon which the colours will be mapped to
+        '''
+        cmin, cmax = self.getVolumeMapWindow((min, max), 'scalar')
+        self.setVolumeColorLevelWindow(cmin, cmax, update_pipeline)
+
+    def setGradientOpacityWindow(self, min, max, update_pipeline=True):
+        '''
+        Parameters
+        -----------
+        min, max: float, default: (80., 99.)
+            the pupper and lower image gradient values that the 
+            opacity will be mapped to if setVolumeRenderOpacityMethod
+            has been set to 'gradient'.
+        update_pipeline: bool
+            whether to immediately update the pipeline with this new
+            setting
+        '''
+        self.gradient_opacity_limits = (min, max)
+        if update_pipeline:
+            self.updateVolumePipeline()
+
+    def getGradientOpacityWindow(self):
+        '''
+        Returns
+        -----------
+        (min, max): tuple, default: (80., 99.)
+            the pupper and lower image gradient values that the 
+            opacity will be mapped to if setVolumeRenderOpacityMethod
+            has been set to 'gradient'.
+        '''
+        return self.gradient_opacity_limits
 
-        return colors, opacity
     
-    def setVolumeColorMapName(self, cmap='magma'):
+    def setScalarOpacityWindow(self, min, max, update_pipeline=True):
+        '''
+        Parameters
+        -----------
+        min, max: float, default: (80., 99.)
+            the upper and lower image values that the 
+            opacity will be mapped to if setVolumeRenderOpacityMethod
+            has been set to 'scalar'.
+        update_pipeline: bool
+            whether to immediately update the pipeline with this new
+            setting
+        '''
+        self.scalar_opacity_limits = (min, max)
+        if update_pipeline:
+            self.updateVolumePipeline()
+
+    def getScalarOpacityWindow(self):
+        '''
+        Returns
+        -----------
+        (min, max): tuple, default: (80., 99.)
+            the upper and lower image values that the 
+            opacity will be mapped to if setVolumeRenderOpacityMethod
+            has been set to 'scalar'.
+        '''
+        return self.scalar_opacity_limits
+
+    def setVolumeColorLevelWindow(self, min, max, update_pipeline=True):
```

Comment:
maybe `setVolumeColorMap` would be more appropriate, or `setVolumeColorMapping`?

Reply:
renamed to setVolumeColorWindow